src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -366,14 +412,47 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
                     scene.pointerY = this._screenCoordinatesRef.y;
                 }
             }
-            controllerData.pick = this._scene.pickWithRay(controllerData.tmpRay, this._scene.pointerMovePredicate || this.raySelectionPredicate);
 
-            const pick = controllerData.pick;
+            let pick = null;
+            let hoverAtOrigin = false;
+            if(controllerData.hoverIndexMesh) {
+                let hoverInfo = this._pickWithMesh(controllerData.hoverIndexMesh, false, this.nearPickPredicate || this.nearGrabPredicate);
+                let nearHover = hoverInfo && hoverInfo.pickedPoint && hoverInfo.hit;
+                if(hoverInfo?.pickedPoint)
+                {
+                    hoverAtOrigin = hoverInfo.pickedPoint.x == 0 && hoverInfo.pickedPoint.y == 0 && hoverInfo.pickedPoint.z == 0;
+                }
+                if(nearHover && !hoverAtOrigin) {                  
+                    controllerData.nearHover = true;
+                    pick = hoverInfo;
+                }
+            }
 
-            if (pick && pick.pickedPoint && pick.hit) {
-                // Update laser state
-                this._updatePointerDistance(controllerData.laserPointer, pick.distance);
+            if(controllerData.pickIndexMesh) {
+                let pickInfo = this._pickWithMesh(controllerData.pickIndexMesh, false, this.nearPickPredicate);
+                let nearPick = pickInfo && pickInfo.pickedPoint && pickInfo.hit;
+                let pickAtOrigin = false;
+                if(pickInfo?.pickedPoint)
+                {
+                    pickAtOrigin = pickInfo.pickedPoint.x == 0 && pickInfo.pickedPoint.y == 0 && pickInfo.pickedPoint.z == 0;
+                }
+                if(nearPick && !pickAtOrigin) {
+                    controllerData.nearPick = true;
+                    pick = pickInfo;
+                }
+            }
 
+            if (!controllerData.nearHover && !controllerData.nearPick && !controllerData.nearGrab) {

```

Comment:
Where are these values ever reset?

Reply:
very good catch! probably they should be reset at the beginning o OnXRFrame. Currently they are not reset.